### ISSUE TITLE
Crash with the file browser when empty selection is active.

### ISSUE THREAD

**Issue by ApostolosB** [Obsidian-StudiosInc/ecrire#24]
to rep
Open ecrire and resize the window. move the mouse cursor up (passing through the first line in order to select it -see the other bug) and hit open. Hit ok. It will crash with:
```
(gdb) bt
#0  0x004d49c2 in _open_file ()
#1  0x004d4c62 in _fs_open_done ()
#2  0x004d74d6 in _cleaning_cb ()
#3  0xb779670f in _eo_evas_smart_cb (data=0x1e59df0, event=0xbf98330c) at lib/evas/canvas/evas_object_smart.c:78
#4  0xb7f1f912 in _event_callback_call (legacy_compare=0 '\000', event_info=0xc6c708, desc=0x1dff2c0, pd=0x1816f68, obj_id=0x8002ae81) at lib/eo/eo_base_class.c:1545
#5  0xb7f1f912 in _efl_object_event_callback_call (obj_id=0x8002ae81, pd=0x1816f68, desc=0x1dff2c0, event_info=0xc6c708) at lib/eo/eo_base_class.c:1606
#6  0xb7f19dd6 in efl_event_callback_call (obj=0x8002ae81, desc=0x1dff2c0, event_info=0xc6c708) at lib/eo/eo_base_class.c:1609
#7  0xb7f19dd6 in efl_event_callback_call (obj=0x8002ae81, desc=0x1dff2c0, event_info=0xc6c708) at lib/eo/eo_base_class.c:1609
#8  0xb7b04fde in _event_to_legacy_call (event_info=, evt_desc=0xb7e5aba8 , obj=0x8002ae81) at lib/elementary/elc_fileselector.c:135
#9  0xb7b04fde in _model_event_call (obj=0x8002ae81, evt_desc=0xb7e5aba8 , model=0x8003ccc1, path=0xc6c708 "/home/toliz")
    at lib/elementary/elc_fileselector.c:141
#10 0xb7b083e1 in _ok (data=0x8002ae81, event=0xbf98350c) at lib/elementary/elc_fileselector.c:1365
#11 0xb7f1e62e in _event_callback_call (legacy_compare=1 '\001', event_info=0x0, desc=0xb6844a2c , pd=0x1e3f798, obj_id=0x8003a067)
    at lib/eo/eo_base_class.c:1545
#12 0xb7f1e62e in _efl_object_event_callback_legacy_call (obj_id=0x8003a067, pd=0x1e3f798, desc=0xb6844a2c , event_info=0x0) at lib/eo/eo_base_class.c:1618
#13 0xb7f19f36 in efl_event_callback_legacy_call (obj=0x8003a067, desc=0xb6844a2c , event_info=0x0) at lib/eo/eo_base_class.c:1621
#14 0xb7f19f36 in efl_event_callback_legacy_call (obj=0x8003a067, desc=0xb6844a2c , event_info=0x0) at lib/eo/eo_base_class.c:1621
#15 0xb7b401a3 in _activate (obj=0x8003a067) at lib/elementary/efl_ui_button.c:73
#16 0xb6ffa8bc in edje_match_callback_exec_check_finals (prop=, ed=0x1e3fc08, source=0x124a160 "elm", sig=0x12ccdd8 "elm,action,click", source_states=, signal_states=, matches=0x1375470, ssp=0xd71790) at lib/edje/edje_match.c:556
#17 0xb6ffa8bc in edje_match_callback_exec (ssp=0xd71790, matches=0x1375470, sig=0x12ccdd8 "elm,action,click", source=0x124a160 "elm", ed=0x1e3fc08, prop=0 '\000')
    at lib/edje/edje_match.c:711
#18 0xb7001b9a in _edje_emit_cb (prop=0 '\000', data=0x0, src=0x124a160 "elm", sig=0x12ccdd8 "elm,action,click", ed=0x1e3fc08) at lib/edje/edje_program.c:1592
#19 0xb7001b9a in _edje_emit_handle (ed=0x1e3fc08, sig=0x12ccdd8 "elm,action,click", src=0x124a160 "elm", sdata=0x0, prop=0 '\000') at lib/edje/edje_program.c:1544
#20 0xb6ffbdd5 in _edje_message_process (em=0x14205c0) at lib/edje/edje_message_queue.c:790
#21 0xb6ffc2e8 in _edje_message_queue_process () at lib/edje/edje_message_queue.c:893
#22 0xb6ffc4cc in _edje_message_queue_process () at lib/edje/edje_message_queue.c:859
#23 0xb6ffc4cc in _edje_job (data=0x0) at lib/edje/edje_message_queue.c:260
#24 0xb7a42aad in _ecore_job_event_handler (data=0x0, type=16, ev=0x139cf80) at lib/ecore/ecore_job.c:98
#25 0xb7a3e266 in _ecore_call_handler_cb (event=, type=, data=, func=) at lib/ecore/ecore_private.h:331
#26 0xb7a3e266 in _ecore_event_call () at lib/ecore/ecore_events.c:629
#27 0xb7a46e47 in _ecore_main_loop_iterate_internal (once_only=once_only@entry=0) at lib/ecore/ecore_main.c:2408
#28 0xb7a47277 in ecore_main_loop_begin () at lib/ecore/ecore_main.c:1308
#29 0x004d6e88 in elm_main ()
#30 0x004d6f18 in main ()
(gdb)

```

**Comment by wltjr** [comment]
Not able to replicate, this maybe ecrire specific. Or triggered by the other issue #21. I think maybe a regression or new issue in git. I may have to try out latest git snapshot and see if I can replicate. I hope something I can address, but sounds like a focus issue and internal like others.... Very frustrating as a developer. Some what regret switching to elm_code... Though I run into tons of EFL and E bugs. I cannot use bryce at all. Stopped me from updating Clipboard to newer modules API, and making a Network Manager one...

**Comment by ApostolosB** [comment]
Network manager is being worked on. Anyway. Can replicate this on both my efl git and stable systems.

**Comment by wltjr** [comment]
Who is working on Network Manager? I am aware of this [one](https://github.com/eyoz/oconnection), but its using the old API. Not sure why someone would start something new using old API. I am not aware of any others underway and surely has not updated the [EFl Apps Todo](https://phab.enlightenment.org/w/efl_apps_todo/) I created. I would not be surprised if another was underway. Seems like working on the same thing together is not something the EFL/E community is into. Like verne, then spawny/anna, and other stuff. Its really odd. Even ephoto is left mostly up to the main author. When they were MIA no one wanted to fix issues till they returned. Thus I would not be surprised on a network manager. Though I am not aware of one underway. If you are thats great, please provide  a link. Thanks!

**Comment by ApostolosB** [comment]
Okra is working on something for nm. Hasn't released anything yet. As for verne and anna the author is busy with focus stuff and his university stuff.

**Comment by wltjr** [comment]
That figures... @okratitan should be working on refining bryce and fixing issues there vs making modules others have showed interest in. I have discussed that with him. Seems most are not interested in something till I mention it. Then everyone wants to work on their own version. I would have a visible repo if it wasn't for the bryce problems. It stopped me dead in my [progress on clipboard](https://github.com/Obsidian-StudiosInc/clipboard) moving to new API. I left off at updating elm_cnp. Though need to go back to emenu vs elm_button. But I expressed interest in working with [upstream on clipboard](https://github.com/thewaiter/Clipboard/issues/15). Though they are focused on Moshka. They created a good deal of modules and stuff for Moshka Bohdi, E community misses out on.

Regarding verne I was meaning more why EFM and not have verne part of E vs an outside project. Or even hosted on git.e.org. Same thing with anna/spawny. No interest in that being official DM/LM for E. Nor interest in Entrance. Others want to spend time they do not have on their concepts. Which would bind it to E and not really allow log in to other stuff....

The community needs almost more work than the code... IMHO

**Comment by ApostolosB** [comment]
The long term plan is for efm to go away. It uses the internal api and this need to go. The sooner the better. Most likely verne is what will replace it Once bushman finds some time and its ready. 

As for bryces i haven't noticed anything weird. But i don't do weird stuff with it.

**Comment by wltjr** [comment]
I can't even use bryce at all. When I first add it, I cannot add gadgets. I have to log out and back in before the right click menu appears. If I go to add anything it freezes and crashes E. Thus for me anything bryce is totally unusable. I even tried without a shelf and any gadets. To see if it was a new + old causing problems together. But I haven't really ever been able to use it reliably, just stuck with shelf sadly.

I have had numerous issues with EFM such that I do not use that. Nor am I a fan of verne ,mostly due to the massive icon the right taking up space. I filed a [bug](https://github.com/marcelhollerbach/verne/issues/31) on that, but doubt it will be addressed. But I do like verne more than efm, and rather see it in E. But there is no discussion of anything along those lines. So not sure if/when Verne will replace EFM in E. It is better than EFM for the most part. Though does ship with some of its own styling in edj file so that messes with custom themes. Have to update vernes theme to match any custom. I really do not like that aspect. Ephoto is better there mostly uses external edj for effects. I doubt ecrire will ever have an external edje file. Maybe for like mobile and desktop UI layouts.

Also not sure if vernes edj is responsible for [UI bugs](https://github.com/marcelhollerbach/verne/issues/32) or if its in EFL, or verne code.

**Comment by okratitan** [comment]
I didn't decide to work on networkmanager because you were.  It's been in the works before you, as when you mentioned it I told you that the wireless gadget was already setup to take in networkmanager support.  As things have progressed and I've maintained my work on gadgets, the next assignment given was to add networkmanager support to wireless.  Trust me. I would much rather have someone else handle the headache of a dbus frontend for networkmanager that has to fit into a framework already existing, i.e. be a backend to wireless gadget.  The community  has many huge projects going on right now.  Wayland, Gadget system and gadget rewrites, Eo among others.  It's no surprise that there are bumps on the road with this many MAJOR projects going on.

**Comment by wltjr** [comment]
@okratitan FYI I have interacted a bit over the years with some network manager devs due to bugs and things encountered in Gentoo. I [helped](https://qiaomuf.wordpress.com/2011/03/08/gentoo-networkmanager-plug-in-now-supports-non-alnum-ssid/) in the past with Gentoo specific network manager port no longer in use Good thing last time there I said nothing about doing a module for EFL. Given I had just ran into a road block on module development all around due to bryce issues. My comments were not meant as an insult. I do not think @okratitan nor I was aware of the other [oconnection](https://github.com/eyoz/oconnection) started this year.

I was not aware of an assignment on a NM project. I was aware of previous work on wireless I think for connman, and was mentioned much of that could be re-used.

The community has to many major projects, causing major issues and none fully complete or usable. Given my issues experienced with bryce. I could not see moving to developing modules for that without resolving major issues with it. I have yet to be able to use it. I cannot even if I wanted replace the shelf with bryce due to issues.

IMHO doing EO should be finished before anything else. The Wayland stuff should break X less and cause less issues for X users. Who seem to be second class now. Despite many cannot run Wayland even if they wanted to. Not just do to bugs but lacking other stuff. Moving clipboard module to elm_cnp. Seems there is little to no support for copy/paste under Wayland. If there is, elm_cnp seems to have little to no wayland support from what I have seen in code. I could be wrong.

The whole gadgets thing is a total mess. I see most support both old and new. I was looking to move Clipboard from old to new, and drop old entirely. But I could not do a thing due to the bryce issues. I guess I could have continued and maybe used it as a gadget on my desktop. Though I may run into other issues not experienced yet there. Seems to be the case for me across the board with many things.

The amount of issues I experience [lead me to this](https://github.com/Obsidian-StudiosInc/os-xtoo#e). Months after my [presentation](https://www.youtube.com/watch?v=jbDpKjznI3M) where even then I said do not run E. Since then the amount of problems I experience has multiplied exponentially...

Now seems a focus on documentation vs fixing stuff. Its crazy.

**Comment by wltjr** [comment]
@okratitan also having read Samsung open source blog. I have no one to thank. No one has helped me with EFL dev. Seems others got more assistance there and other

"The creator of Ewl, Nathan Ingersoll, took me under his wing and began teaching me C and how to use Ewl and other Enlightenment Foundation Libraries (EFL)."
https://blogs.s-osg.org/introducing-stephen-houston-newest-intern/

"Carsten Haitzler (Rasterman) guided me through the process of writing the code for this project."
https://blogs.s-osg.org/samsung-osg-internship-program-marcel-hollerbach/

Nothing bad, just saying. Rather than help, I feel I have gotten the opposite. Like how to customize text. Which was trivial and took me a month to figure out even asking others... As just mentioned on list... I also interact with many communities. This is not my experience across the board.

Given what all I am doing with no help, and tons of resistance along the way... Not to mention I did jump through requested hoola hoops. Like [joining](https://marc.info/?l=enlightenment-devel&m=149279045530732&w=2) the mailing list per raster request to talk to tom about [taking over ecrire](https://marc.info/?l=enlightenment-devel&m=149437147923347&w=2). Which raster totally forgot about, after requesting I do that. They still [poke at their old buggy ecrire](https://git.enlightenment.org/apps/ecrire.git/)...

**Comment by ApostolosB** [comment]
@wltjr Tested adding a bryce and works like a charm (wayland stable&git e/efl). If its something with X i don't know, i don't use it anymore. The only think i can suggest is backing up .e and .elementary and see it it will work - if you haven't done this already.

Regarding copy paste there is only one issue i am aware of (T6284 on phab) and other than that it works for everything i copy pasted even between xwayland and wayland apps. Maybe from a dev POV things look different but as an end user that's what i experience.

ON verne there is a setting somewhere that disables the whole right hand column but don't remember where and cant look because verne is broken at the  moment. Marcel know but has to fix the focus stuff first.

**Comment by wltjr** [comment]
I can try nuking that. I have a pretty clean one as when I have gone back to 0.21.0 it forced me back to that. But eet does have some major bugs. I tried with 0.22.1. I had to revert my desktop back to 0.21.10 due to daily lock ups. I still have windows that do not render at times under my laptop requiring me to restart E from main menu. 0.22.x has so many issues for me. I doubt its config related but I can try that... It would be great if it was config related. I tend to run into issues others do not... Not sure why.

Did you see the new copy and paste bug filed? #24

I like the right hand column information just not the icon. I have to scroll down to view useful info. I can see about disabling the right hand column. Thank you for that suggestion. Yes focus fix is more important. That effects more stuff. I just would like to see verne replace EFM. Or others show interest there :)
